### PR TITLE
Add 'neo-layout' keyboard layout

### PR DESCRIPTION
**Pull request by lennartS**
More information about the layout (including a picture) can be found on https://neo-layout.org or on
https://en.wikipedia.org/wiki/Neo_(keyboard_layout)

### CODE REVIEW

File: luaui/configs/keyboard_layouts.lua [hunk 1]
```diff
@@ -76,7 +76,47 @@ scanToCode["dvorak"] = {
...
+	[";"] = "d",
+	["'"] = "y",
+	[","] = ",",
+	["."] = ".",
+	["/"] = "j",
+	["`"] = "^",
+	["-"] = "-",
+	["="] = "`",
+	-- NEEDS CORRECTION BELOW
+	["\\"] = "\\",
```
Comment (badosu) [inline luaui/configs/keyboard_layouts.lua:115]: If this is correct you don't need the comment above and below

Comment (lennartS) [reply luaui/configs/keyboard_layouts.lua:115]: Thanks for the review!

It's not correct and it's rather tricky to correct this. The key which is on qwerty used for the `\` is used as modifier for the third level on NEO. So it's not even a dead key like the accents, but just triggers a change for all other keys.

I'm not sure how to properly implement this in the keymap.

**Comment by lennartS** [comment]
Sorry for the drive-by commit. I don't use Discord, and the Subreddit didn't seem to be an appropriate place to start a discussion, so I just opened the PR.

File: luaui/configs/keyboard_layouts.lua (hunk 1 above)
Comment (badosu) [reply luaui/configs/keyboard_layouts.lua:115]: I see, thank you for clarification.

**Comment by badosu** [comment]
LGTM :+1:

**Comment by lennartS** [comment]
I added the comment to the source directly, so this information is not just kept in the PR.

File: luaui/configs/keyboard_layouts.lua (hunk 1 above)
Comment (badosu) [reply luaui/configs/keyboard_layouts.lua:119]: Keep in mind these layouts are just a workaround for engine not having translation layer from scancode -> key label yet (used only for presentation purposes)

So it's not a big deal.

File: luaui/configs/keyboard_layouts.lua [hunk 2]
```diff
@@ -76,7 +76,51 @@ scanToCode["dvorak"] = {
...
+	Z = "Ü",
+	X = "Ö",
+	C = "Ä",
+	V = "P",
+	B = "Z",
+	N = "B",
+	M = "M",
+	[";"] = "d",
+	["'"] = "y",
+	[","] = ",",
```
Comment (badosu) [inline luaui/configs/keyboard_layouts.lua:108]: It's likely this one requires `["comma"]` instead.

You can test it by doing something like:

```
/unbindkeyset sc_comma
/bind sc_comma stop
```

Then hovering over the stop button in order menu and seeing what the tooltip shows

Comment (lennartS) [reply luaui/configs/keyboard_layouts.lua:108]: If I put in the commands you listed, the tooltip of the stop order still shows 'O' (capital letter o), just as before. It doesn't change the tooltip.

Comment (badosu) [reply luaui/configs/keyboard_layouts.lua:108]: requires toggling the widget on and off to pick up the update

**Pull request merged**